ibis/expr/api.py

```diff
@@ -383,55 +379,23 @@ def table(
     """
     if schema is not None:
         schema = sch.schema(schema)
-    return _table(schema, name, data, columns)
-
-
-_table = Dispatcher("_table")
-
-
-@_table.register(sch.Schema, (str, type(None)), type(None), type(None))
-def _unbound_table(schema, name, *_) -> Table:
     return ops.UnboundTable(schema=schema, name=name).to_expr()
 
 
-@_table.register(
-    (sch.Schema, type(None)), (str, type(None)), pd.DataFrame, type(None)
-)
-def _data_frame(schema, name, data, _) -> Table:
-    if schema is None:
-        schema = sch.infer(data)
-    return ops.PandasTable(name=name, schema=schema, data=data).to_expr()
-
-
-@_table.register(
-    type(None),
-    (str, type(None)),
-    collections.abc.Sequence,
-    type(None),
-)
-def _sequence_of_rows_no_columns(_, name, data, columns) -> Table:
-    return _table(sch.infer(data), name, data, columns)
-
-
-@_table.register(
-    type(None),
-    (str, type(None)),
-    collections.abc.Sequence,
-    collections.abc.Sequence,
-)
-def _sequence_of_rows(_, name, data, columns) -> Table:
-    schema = sch.Schema(columns, sch.infer(data).types)
-    return _table(schema, name, data, None)
-
-
-@_table.register(
-    sch.Schema,
-    (str, type(None)),
-    collections.abc.Sequence,
-    type(None),
-)
-def _sequence_of_rows_with_schema(schema, name, data, _) -> Table:
-    return ops.PythonTable(name=name, schema=schema, data=data).to_expr()
+# @functools.singledispatch
+# def dataset(data, schema=None, name=None):
```

Comment:
We *could* have a different API for constructing file or in memory based tables, perhaps dataset would be a good name for it?

Reply:
I'd really like this be called `table`. `dataset` isn't an obvious name IMO.

That said, I'd be okay with calling it `table_data`, `table_literal`, or even `memtable`.

Reply:
I kinda like `ibis.memtable()`, what API would we use for single files and remote datasets?